CargoMonitor/CargoMonitor.cs

```diff
@@ -294,25 +300,24 @@ private bool _handleCommodityCollectedEvent(CommodityCollectedEvent @event)
             Cargo cargo = GetCargoWithEDName(@event.commodityDefinition?.edname);
             if (cargo != null)
             {
+                Haulage haulage = cargo.haulageData.FirstOrDefault(h => h.missionid == @event.missionid);
```

Comment:
Can this be 'SingleOrDefault'?
Or can the collection contain multiple items with the same 'missionId'?

Reply:
The general pattern Hooda appeared to follow was to try to look up haulage, then only add it if it was missing. Consequently, I don't think it's intended to contain multiple items with the same missionId but you're right that it isn't designed to prevent it. I have tried to write my changes to preclude that possibility, but since there is no duplicate protection built in I think that swapping to use 'SingleOrDefault' does create additional risk.

The same is true for cargo in the inventory collection.

We could consider creating a separate ticket to transition from lists to dictionaries for both the cargo in inventory and for haulage (using commodity `edname` as a unique key for cargo and haulage `missionid` as a unique key for haulage) but doing so could have implications for other dependent projects and is would have a wider impact than I intended for this first pass.

Reply:
I would prefer to leave it as is for now.